### PR TITLE
feat(ln): add libp2p transport flags to config

### PR DESCRIPTION
**Pull request by clearloop**
## Summary

 - [x] fix gossipsub config
 - [x] update deprecated implementations
 - [x] introduce new config for lighthouse network that can switch libp2p strategy
   - tcp
   - nym
   - tcp or nym
 - [x] build transport via libp2p strategy

### CODE REVIEW

File: beacon_node/lighthouse_network/src/service/mod.rs [hunk 1]
```diff
@@ -328,9 +321,13 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
...
-            let (transport, bandwidth) = build_transport(local_keypair.clone())
-                .await
-                .map_err(|e| format!("Failed to build transport: {:?}", e))?;
+            let (transport, bandwidth) = match config.libp2p_strategy {
+                Libp2pStrategy::Nym => build_nym_transport(local_keypair.clone()).await,
+                Libp2pStrategy::Tcp => build_tcp_transport(local_keypair.clone()).await,
+                Libp2pStrategy::NymOrTcp => build_transport(local_keypair.clone()).await,
+            }
+            .map_err(|e| format!("Failed to build transport: {:?}", e))?
+            .with_bandwidth_logging();
```
Comment (clearloop) [inline beacon_node/lighthouse_network/src/service/mod.rs]: now we can build the transport with our network config but not compilation flags ( features )

File: beacon_node/lighthouse_network/src/config.rs [hunk 2]
```diff
@@ -52,6 +52,13 @@ pub fn gossip_max_size(is_merge_enabled: bool) -> usize {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum Libp2pStrategy {
+    Nym,
+    Tcp,
+    NymOrTcp,
```
Comment (clearloop) [inline beacon_node/lighthouse_network/src/config.rs]: use `Auto` ?

File: beacon_node/lighthouse_network/src/config.rs [hunk 3]
```diff
@@ -52,6 +52,13 @@ pub fn gossip_max_size(is_merge_enabled: bool) -> usize {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum Libp2pStrategy {
```
Comment (clearloop) [inline beacon_node/lighthouse_network/src/config.rs]: proper naming?

Comment (nulltea) [reply beacon_node/lighthouse_network/src/config.rs]: `Strategy` part is quite ambiguous, `Libp2pTransport` would be more descriptive IMO

File: beacon_node/lighthouse_network/src/service/mod.rs (hunk 1 above)
Comment (nulltea) [reply beacon_node/lighthouse_network/src/service/mod.rs]: great, ty!

File: beacon_node/lighthouse_network/src/config.rs (hunk 2 above)
Comment (nulltea) [reply beacon_node/lighthouse_network/src/config.rs]: perhaps `Either` as in Libp2p terminology

**Review by nulltea — APPROVED** [review]
Looks good, but consider changing enum names where discussed

**Pull request merged**